misc/pkg.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+import sys
+import os
+import logging
+import logging.handlers
+import subprocess
+import argparse
+import tempfile
+import os.path
+import json
+import glob
+import shutil
+import re
+
+fmt_string='%(levelname)s:%(asctime)s %(filename)s:%(lineno)d] %(message)s'
+logging.basicConfig(format=fmt_string)
+formatter = logging.Formatter(fmt_string)
+for h in logging.getLogger().handlers: h.setFormatter(formatter)
+logger = logging.getLogger('pkg')
+logger.setLevel(logging.DEBUG)
+
+
+BOOTSTRAP_ENVIRONMENT = b"""
+import os
+import sys
+__entry_point__ = None
+if '__file__' in locals() and __file__ is not None:
+  __entry_point__ = os.path.dirname(__file__)
+elif '__loader__' in locals():
+  from zipimport import zipimporter
+  from pkgutil import ImpLoader
+  if hasattr(__loader__, 'archive'):
+    __entry_point__ = __loader__.archive
+  elif isinstance(__loader__, ImpLoader):
+    __entry_point__ = os.path.dirname(__loader__.get_filename())
+if __entry_point__ is None:
+  sys.stderr.write('Could not launch python executable!\\n')
+  sys.exit(2)
+sys.path[0] = os.path.abspath(sys.path[0])
+sys.path.insert(0, os.path.abspath(os.path.join(__entry_point__, '.bootstrap')))
+from _pex.pex_bootstrapper import bootstrap_pex
+bootstrap_pex(__entry_point__)
+"""
+
+RUN_TEST=b"""
+import os
+import sys
+import subprocess
+def run_subprocess(cmd):
+    proc =  subprocess.Popen(
+        cmd,
+        stdout=sys.stdout,
+        stderr=sys.stderr,
+        env=environ,
+        shell=True
+    )
+    return_code = 0
+
+    try:
+        return_code = proc.wait()
+        sys.stdout.flush()
+        sys.stderr.flush()
+    except Exception as e:
+        logger.exception("Could not run command: % ", e)
+        proc.terminate()
+        raise
+
+"""
+
+def get_ldd_lines(binary):
+    ret = str(subprocess.check_output("ldd {}".format(binary), shell=True))
```

Comment:
also might not be needed if we use cmake packs

https://cmake.org/Wiki/CMake:CPackPackageGenerators#RPM_.28Unix_Only.29